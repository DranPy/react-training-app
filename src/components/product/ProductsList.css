.products-list ul {
  display: flex;
  flex-wrap: wrap;
}

.products-list h1 {
  position: relative;
  padding-right: 40px;
}

.products-list  h1 .products-list__btn-add {
 position: absolute;
 right: 0;
 top: 5px; 
}


.product-item  {
  margin-bottom: 10px;
}

.product-item .product-item__name {
  margin-top: 10px;
}

.product-item .product-item__price {
  margin-bottom: 10px;
}
