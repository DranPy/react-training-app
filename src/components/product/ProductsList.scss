.products-list {
  ul { 
    display: flex;
    flex-wrap: wrap;
  }

  header { 
    display: flex;
    align-items: center;
  }

  &__btn-add {
    margin-left: auto;
  }
}

.product-item {
  margin-bottom: 10px;

  &__name {
    margin-top: 10px;
  }

  &__price {
    margin-bottom: 10px;
  }
}