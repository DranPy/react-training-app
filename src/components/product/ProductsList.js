import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ProductItem from './ProductItem';
import Loader from '../Loader';
import { FontAwesomeIcon as FA } from '@fortawesome/react-fontawesome';
import { Link } from 'react-router-dom';

import './ProductsList.css';

class ProductsList extends Component {
  static propTypes = {
    products: PropTypes.array.isRequired,
    fetchProducts: PropTypes.func.isRequired,
    deleteProduct: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired,
  };

  componentDidMount() {
    this.props.fetchProducts();
  }

  render() {
    const { isLoading, products, deleteProduct } = this.props;

    return (
      <div className="products-list">
        <div className="container">
          <h1>
            Products
            <Link
              to="/products/add"
              className="products-list__btn-add btn btn-success rounded-circle"
            >
              <FA icon="plus" />
            </Link>
          </h1>
          <ul className="row list-unstyled">
            {!isLoading && products ? (
              products.map(product => (
                <ProductItem key={product.id} onDelete={deleteProduct} product={product} />
              ))
            ) : (
              <Loader />
            )}
          </ul>
        </div>
      </div>
    );
  }
}

export default ProductsList;
