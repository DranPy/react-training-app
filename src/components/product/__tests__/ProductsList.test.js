import { shallow } from 'enzyme';
import React from 'react';

import ProductsList from '../ProductsList';

describe('<ProductsList />', () => {
  const defaultProps = {
    fetchProducts: () => {},
    isLoading: false,
    products: [
      {
        id: '1',
        name: 'test1',
        category: 'category 1',
        price: '1$',
        imgUrl: 'src/1.jpg',
      },
    ],
  };

  it('should render component', () => {
    const wrapper = shallow(<ProductsList {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render component', () => {
    const wrapper = shallow(<ProductsList {...defaultProps} isLoading={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
