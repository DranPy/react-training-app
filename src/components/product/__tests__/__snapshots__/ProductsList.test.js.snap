// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsList /> should not render component 1`] = `
<div
  className="products-list"
>
  <div
    className="container"
  >
    <h1>
      Products
      <button
        className="products-list__btn-add btn btn-success rounded-circle"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="plus"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </h1>
    <ul
      className="row list-unstyled"
    >
      <Loader />
    </ul>
  </div>
</div>
`;

exports[`<ProductsList /> should render component 1`] = `
<div
  className="products-list"
>
  <div
    className="container"
  >
    <h1>
      Products
      <button
        className="products-list__btn-add btn btn-success rounded-circle"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="plus"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </h1>
    <ul
      className="row list-unstyled"
    >
      <ProductItem
        key="1"
        product={
          Object {
            "category": "category 1",
            "id": "1",
            "imgUrl": "src/1.jpg",
            "name": "test1",
            "price": "1$",
          }
        }
      />
    </ul>
  </div>
</div>
`;
